Add org-sliced-images

jcfk opened the pull request:
### Brief summary of what the package does

Display images as sliced images in org-mode, just like `insert-sliced-image`. This makes scrolling nicer. It overrides a number of the default image-display functions.

### Direct link to the package repository

https://github.com/jcfk/org-sliced-images

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this. Have you compared against https://melpa.org/#/iscroll ?

In the meantime, quick first pass --

`org-sliced-images.el` with byte-compile using Emacs 29.2:
```
org-sliced-images.el:226:51: Warning: the function `org-attach-expand' is not known to be defined.
org-sliced-images.el:187:31: Warning: the function `org-element-property' is not known to be defined.
org-sliced-images.el:184:27: Warning: the function `org-element-lineage' is not known to be defined.
```

`org-sliced-images.el` with checkdoc 0.6.2 (fix *within reason*):
```
org-sliced-images.el:284: Argument `ov' should appear (as OV) in the doc string
```

`org-sliced-images.el` with package-lint 20240220.2303:
```
1 issue found:
8:38: warning: Use a properly versioned dependency on "org" if possible.
```

jcfk commented:
Thank you for the comments. Yes, I have compared it to iscroll, and I believe my package differs in two ways:

- iscroll achieves sliced scrolling by manually calculating set-window-vscroll amounts with each movement operation. This results in certain bugs when it comes to tall images, which I have reproduced on my end: casouri/iscroll#3.
- For the above reason, iscroll is also more resource-intensive, to the point of making a graphical difference when scrolling fast on my machine.

org-sliced-images inserts images as actual slices during the display process. Scrolling over these is therefore as "native" as one could expect, and one can use all the ordinary movements (no special config for evil, etc).

I have fixed all the issues and warnings and added an iscroll comparison section to the readme.

riscy commented:
Thank you for the thoughtful reply.  I buy it and you have a novel solution.

This is not a blocker, but your commentary says "See homepage for more information" and your summary line says `Sliced inline images in org` - users browsing the MELPA index will have no idea what this is.  I would just take what you've written above under `Brief summary of what the package does` and make that your `;;; Commentary`.  Your summary line could be `Improve image scrolling performance in org` or similar.
